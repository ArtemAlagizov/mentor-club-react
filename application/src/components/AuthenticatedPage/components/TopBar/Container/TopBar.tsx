import React from "react";
import TopBarView from "../View/TopBarView";
import { ITopBar } from "./ITopBar";

const TopBar: React.FC<ITopBar> = ({ logout }) => {
  return <TopBarView logout={logout} />;
};

export default TopBar;
