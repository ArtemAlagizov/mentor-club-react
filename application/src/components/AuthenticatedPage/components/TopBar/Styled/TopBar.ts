import styled from "styled-components";

const buttonColor = "#0e0f16";

export const TopBar = styled.header`
  /* stylelint-disable-next-line */
  display: flex;
  justify-content: space-between;
  padding: 0 20px;
  height: 4rem;
  width: 100%;
  flex-shrink: 0;
  background: ${buttonColor};
`;
