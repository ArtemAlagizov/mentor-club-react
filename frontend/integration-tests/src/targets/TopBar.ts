/// <reference types="Cypress" />

export class TopBar {
  public static getProfilePopOver = () =>
    cy.get('[data-id="top-bar-profile-image"]').parent();

  public static getLogoutButton = () =>
    cy.get('[data-id="top-bar-logout-button"]');
}
