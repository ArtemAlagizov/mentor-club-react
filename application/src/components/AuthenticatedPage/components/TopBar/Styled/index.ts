import { TopBar } from "./TopBar";
import { TopBarTitle } from "./TopBarTitle";
import { UserImageContainer } from "./UserImageContainer";
import { UserImage } from "./UserImage";
import { UserName } from "./UserName";
import { LogoutButton } from "./LogoutButton";

export const Styled = {
  TopBar,
  TopBarTitle,
  UserName,
  UserImage,
  UserImageContainer,
  LogoutButton
};
