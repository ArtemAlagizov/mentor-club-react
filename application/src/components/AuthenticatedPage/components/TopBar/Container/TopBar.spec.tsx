import React from "react";
import { shallow } from "enzyme";
import TopBar from "./TopBar";

describe("Given a TopBar", () => {
  describe("when rendering", () => {
    it("should have TopBar matching snapshot", () => {
      const topBar = shallow(<TopBar logout={jest.fn()} />);

      expect(topBar).toMatchSnapshot();
    });
  });
});
