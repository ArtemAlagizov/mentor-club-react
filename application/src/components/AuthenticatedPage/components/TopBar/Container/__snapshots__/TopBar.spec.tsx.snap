// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a TopBar when rendering should have TopBar matching snapshot 1`] = `
<TopBarView
  logout={[MockFunction]}
/>
`;
